### ISSUE TITLE
[DX] [D8] Add a `validate()` method to entity classes

### ISSUE THREAD

**Issue by avpaderno** [backdrop/backdrop-issues#6061]
In Drupal 8, classes that are used in order to implement a content entity (which extend the [`ContentEntityBase`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Entity%21ContentEntityBase.php/class/ContentEntityBase/8.9.x) class) have a [`validate()`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Entity%21ContentEntityBase.php/function/ContentEntityBase%3A%3Avalidate/8.9.x) method that validates the values entered when creating the entity.

A similar method should be introduced in Backdrop and added to [`EntityInterface`](https://docs.backdropcms.org/api/backdrop/core%21modules%21entity%21entity.class.inc/interface/EntityInterface/1).

**Comment by kiamlaluno** [comment]
This is probably a Backdrop 2.x feature. Contributed modules could have implemented a `validate()` method in their entities that could be not compatible with the implementation Backdrop adds.

**Comment by argiepiano** [comment]
> This is probably a Backdrop 2.x feature.

Agreed. This would be a fairly complex and deep change to the way Backdrop deals with entities and data in general (e.g. fields). It's not just about creating a `validate` method for the entity class. The Drupal `ContentEntityBase::validate()` method actually calls the validate method for the entity's  "typed data" objects - for example, it's fields and properties.
